### PR TITLE
fixed tests

### PR DESCRIPTION
**Pull request by robertsLando**

### CODE REVIEW

**Comment by robertsLando** [comment]
@mcollina Actually I have disabled clusters on tests, I can confirm that with clusters tests are not working only with mongodb, seems they are falky, sometimes work sometimes not. It's like mongo is missing some messages 

If I run mongodb test using mqemitter-redis instead mqemitter-mongo they works.

In poor words I think that `mqemitter-mongodb` is not working as expected with clusters

**Comment by robertsLando** [comment]
By enabling logs:

```
# Subtest: Subscribed clients receive updates - QoS 1
    1..10 # each client should receive a message
BROKER_8: Client Connected: subscriber_6 to broker
BROKER_7: Client Connected: subscriber_5 to broker
BROKER_1: Client Connected: subscriber_3 to broker
BROKER_2: Client Connected: subscriber_4 to broker
BROKER_4: Client Connected: subscriber_2 to broker
BROKER_3: Client Connected: subscriber_0 to broker
BROKER_5: Client Connected: subscriber_1 to broker
BROKER_6: Client Connected: subscriber_7 to broker
BROKER_5: Client BROKER_BROKER_5 has published subscriber_1 on $SYS/BROKER_5/new/clients
BROKER_3: Client BROKER_BROKER_3 has published subscriber_0 on $SYS/BROKER_3/new/clients
BROKER_3: Client Connected: subscriber_8 to broker
BROKER_1: Client BROKER_BROKER_1 has published subscriber_3 on $SYS/BROKER_1/new/clients
BROKER_4: Client Connected: subscriber_9 to broker
BROKER_6: Client BROKER_BROKER_6 has published subscriber_7 on $SYS/BROKER_6/new/clients
BROKER_3: Client BROKER_BROKER_3 has published subscriber_8 on $SYS/BROKER_3/new/clients
BROKER_5: MQTT client subscriber_1 subscribed to topics: subscribers/topic
BROKER_3: MQTT client subscriber_0 subscribed to topics: subscribers/topic
BROKER_4: MQTT client subscriber_2 subscribed to topics: subscribers/topic
BROKER_1: MQTT client subscriber_3 subscribed to topics: subscribers/topic
BROKER_6: MQTT client subscriber_7 subscribed to topics: subscribers/topic
BROKER_7: MQTT client subscriber_5 subscribed to topics: subscribers/topic
BROKER_3: MQTT client subscriber_8 subscribed to topics: subscribers/topic
BROKER_4: MQTT client subscriber_9 subscribed to topics: subscribers/topic
BROKER_4: Client BROKER_BROKER_4 has published subscriber_2 on $SYS/BROKER_4/new/clients
BROKER_7: Client BROKER_BROKER_7 has published subscriber_5 on $SYS/BROKER_7/new/clients
BROKER_4: Client BROKER_BROKER_4 has published subscriber_9 on $SYS/BROKER_4/new/clients
BROKER_8: Client BROKER_BROKER_8 has published subscriber_6 on $SYS/BROKER_8/new/clients
BROKER_2: Client BROKER_BROKER_2 has published subscriber_4 on $SYS/BROKER_2/new/clients
BROKER_2: MQTT client subscriber_4 subscribed to topics: subscribers/topic
BROKER_8: MQTT client subscriber_6 subscribed to topics: subscribers/topic
BROKER_1: Client Connected: mqttjs_aaa6fa50 to broker
BROKER_1: Client BROKER_BROKER_1 has published mqttjs_aaa6fa50 on $SYS/BROKER_1/new/clients
    ok 1 - Message received from subscriber_1
    ok 2 - Message received from subscriber_0
    ok 3 - Message received from subscriber_8
    ok 4 - Message received from subscriber_7
    ok 5 - Message received from subscriber_3
    ok 6 - Message received from subscriber_6
BROKER_1: Client mqttjs_aaa6fa50 has published Hello world on subscribers/topic
```

**Comment by robertsLando** [comment]
@mcollina finally seems everything is working :+1:  Some tests fails just because are flaky

**Comment by mcollina** [comment]
> Some tests fails just because are flaky

Why? We might want to fix them.

**Comment by robertsLando** [comment]
@mcollina This has become a personal challenge for every bug I fix I found another one :rofl: 

Hopefully it's working now

UPDATE: Mongodb still failing (everthing is working on my instance). I need to understand why...

**Comment by robertsLando** [comment]
@mcollina Do you know where I can find some kind of mqtt benchmark tool written in nodejs? I would like to integrate benchmarks in this repo, start aedes like I do now in a separete process and run the benchmark against it with different persistences and with/without clusters to see how it performs

File: test/basic.test.js [hunk 1]
```diff
@@ -28,8 +28,25 @@ test('Connect-Subscribe-Publish-Disconnect 300 clients using WS and MQTT/MQTTS p
   await pMap(clients, c => c.end(), pMapOptions)
 })
 
+test('Unhautorized client', function (t) {
+  t.tearDown(helper.closeBroker)
+  t.plan(1)
+
+  const noError = t.error.bind(t)
```
Comment (mcollina) [inline test/basic.test.js]: You can just use async/await.

Comment (robertsLando) [reply test/basic.test.js]: I have problems when using async functions with test, they end the test when they resolve and so even if I have a t.plan they doesn't wait for that to call end. In this case I have to setup a listener and without that it doesn't work

Comment (mcollina) [reply test/basic.test.js]: This should work:

```js
await helper.startBroker()
var client = helper.startClient('mqtt', { username: 'user', password: 'notallowed' }, true)
var err = await once(client, 'error')
t.equal(err.code, 4, 'Connection should be rejected with code 4')
```

Comment (robertsLando) [reply test/basic.test.js]: You always tech me something new :fearful:

Comment (robertsLando) [reply test/basic.test.js:35]: @mcollina Where do I find `once`? `events`? becaus it isn't working

File: test/basic.test.js [hunk 2]
```diff
@@ -54,49 +68,64 @@ async function testQos (t, qos) {
...
-  await pMap(subscribers, c => c.end(), pMapOptions)
-  await publisher.end()
+  await publisher.publish(msg.topic, msg.payload, msg)
 }
 
-test('Subscribed clients receive updates - QoS 1', async function (t) {
-  await testQos(t, 1)
+test('Subscribed clients receive updates - QoS 1', function (t) {
+  testQos(t, 1)
+    .catch(t.error.bind(t))
```
Comment (mcollina) [inline test/basic.test.js:95]: why this is not using await anymore?

**Review by mcollina — APPROVED** [review]
CI is still failing, but code is ok

**Comment by robertsLando** [comment]
For the same reason I told you before... tests were flacky in previous version

---

Daniel - Software Engineer

Support me at:
    Github sponsors

File: test/basic.test.js (hunk 2 above)
Comment (robertsLando) [reply test/basic.test.js:95]: For the same reason I told you before... tests were flacky in previous version

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**